core/src/main/java/com/linecorp/armeria/common/multipart/DefaultMultipart.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.multipart;
+
+import static com.linecorp.armeria.common.multipart.MultipartEncoder.EMPTY_OPTIONS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import javax.annotation.Nullable;
+
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+
+import com.google.common.base.MoreObjects;
+import com.spotify.futures.CompletableFutures;
+
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.stream.StreamMessage;
+import com.linecorp.armeria.common.stream.SubscriptionOption;
+import com.linecorp.armeria.common.util.UnmodifiableFuture;
+import com.linecorp.armeria.internal.common.HttpObjectAggregator;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.util.concurrent.EventExecutor;
+
+final class DefaultMultipart implements Multipart {
+
+    /**
+     * The default boundary used for encoding multipart messages.
+     */
+    static final String DEFAULT_BOUNDARY = "!@==boundary==@!";
```

Comment:
Could we use something less likely to collide? For example, if unspecified, we could generate a Base64 representation of a random byte array.

Reply:
That is a good suggestion!